bpf/process/bpf_execve_bprm_check.c

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright Authors of Cilium */
+
+#include "vmlinux.h"
+#include "api.h"
+#include "bpf_tracing.h"
+
+#include "bpf_events.h"
+
+char _license[] __attribute__((section("license"), used)) = "GPL";
+
+__attribute__((section("kprobe/security_bprm_check"), used)) int
+BPF_KPROBE(tg_kp_bprm_check, struct linux_binprm *bprm, int ret)
+{
+	__u32 pid, zero = 0;
+	unsigned int links;
+	struct execve_map_value *curr;
+	struct execve_heap *heap;
+	struct file *file;
+
+	pid = (get_current_pid_tgid() >> 32);
+
+	curr = execve_map_get(pid);
+	if (!curr)
+		return 0;
+
+	heap = map_lookup_elem(&execve_heap, &zero);
+	if (!heap)
+		return 0;
+
+	probe_read(&file, sizeof(file), _(&bprm->file));
+	links = BPF_CORE_READ(file, f_path.dentry, d_inode, __i_nlink);
+	heap->info.inode.i_nlink = links;
+	heap->info.inode.initialized = 1;
+
+	execve_info_map_set(pid, &heap->info);
```

Comment:
should this be the full pid+tgid? Or does it matter. Wonder if you have multiple threads all doing execs does this get confused?

Reply:
Good catch! sorry missed that somehow... So let's make it simple I made it full tid, and commented the fact that execve is complex that can fail at any point and we don't care about kernel locking or thread races between... all may execute our hooks but only the leader will send us the final execve tracepoint which we care for this case.

Leader will get cleaned properly, other threads entries will be just re-used...